text/0431-sagemaker-l2-endpoint.md

```diff
@@ -0,0 +1,1150 @@
+    /**
+     * The model to host.
+     */
+    readonly model: IModel;
+    /**
+     * Name of the production variant.
+     */
+    readonly variantName: string;
+  }
+  ```
+
+- `ProductionVariant` -- represents a production variant that has been associated with an
+  `EndpointConfig`
+
+  ```ts
+  export interface ProductionVariant {
+    /**
+     * The size of the Elastic Inference (EI) instance to use for the production variant. EI instances
+     * provide on-demand GPU computing for inference.
+     *
+     * @default none
+     */
+    readonly acceleratorType?: AcceleratorType;
+    /**
+     * Number of instances to launch initially.
+     */
+    readonly initialInstanceCount: number;
+    /**
+     * Determines initial traffic distribution among all of the models that you specify in the
+     * endpoint configuration. The traffic to a production variant is determined by the ratio of the
+     * variant weight to the sum of all variant weight values across all production variants.
+     */
+    readonly initialVariantWeight: number;
+    /**
+     * Instance type of the production variant.
+     */
+    readonly instanceType: ec2.InstanceType;
+    /**
+     * The name of the model to host.
+     */
+    readonly modelName: string;
+    /**
+     * The name of the production variant.
+     */
+    readonly variantName: string;
+  }
+  ```
+
+- `AcceleratorType` -- an enumeration of values representing the size of the Elastic Inference (EI)
+  instance to use for the production variant. EI instances provide on-demand GPU computing for
+  inference
+
+  ```ts
+  export enum AcceleratorType {
+    /**
+     * Medium accelerator type.
+     */
+    MEDIUM = 'ml.eia1.medium',
+    /**
+     * Large accelerator type.
+     */
+    LARGE = 'ml.eia1.large ',
+    /**
+     * Extra large accelerator type.
+     */
+    XLARGE = 'ml.eia1.xlarge',
+  }
+  ```
+
+#### Endpoint
+
+- `IEndpoint` -- the interface for a SageMaker Endpoint resource
+
+  ```ts
+  export interface IEndpoint extends cdk.IResource {
+    /**
+     * The ARN of the endpoint.
+     *
+     * @attribute
+     */
+    readonly endpointArn: string;
+    /**
+     * The name of the endpoint.
+     *
+     * @attribute
+     */
+    readonly endpointName: string;
+
+    /**
+     * Permits an IAM principal to invoke this endpoint
+     * @param grantee The principal to grant access to
+     */
+    grantInvoke(grantee: iam.IGrantable): iam.Grant;
+  }
+  ```
+
+- `EndpointProps` -- construction properties for a SageMaker endpoint
+
+  ```ts
+  export interface EndpointProps {
+
+    /**
+     * Name of the endpoint.
+     *
+     * @default AWS CloudFormation generates a unique physical ID and uses that ID for the endpoint's
+     * name.
+     */
+    readonly endpointName?: string;
+
+    /**
+     * The endpoint configuration to use for this endpoint.
+     *
+     * [disable-awslint:ref-via-interface]
+     */
+    readonly endpointConfig: EndpointConfig;
```

Comment:
Why is this not an interface?

Reply:
I think this should take an `IEndpointConfig` as is the traditional CDK pattern. 

the implementation for `public get instanceProductionVariants(): IEndpointInstanceProductionVariant[] {` should take this into consideration and throw an error if the EndpointConfig is imported. 

This seems reasonable to me - you can create an `Endpoint` from an imported `EndpointConfig`, but you can't actually reference anything from within the config. Otherwise, there is no point in differentiating an `Endpoint` from an `EndpointConfig`, because users will be importing configs from elsewhere to reuse the config construct.